### PR TITLE
Add DIA SDK sample

### PR DESCRIPTION
**Pull request by riverar**

### CODE REVIEW

File: dia/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        CoInitializeEx(std::ptr::null_mut(), COINIT_MULTITHREADED)?;
+        let dia_source: IDiaDataSource = CoCreateInstance(&DiaSource, None, CLSCTX_INPROC_SERVER)?;
+
+        let mut vec = std::env::current_exe()
+            .unwrap()
+            .as_os_str()
+            .encode_wide()
+            .collect::<Vec<u16>>();
+        vec.push(0x00);
+        vec.push(0x00);
```
Comment (kennykerr) [inline dia/src/main.rs:21]: double termination?

File: dia/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        dia_source.loadDataForExe(PWSTR(vec.as_mut_ptr()), None, None)?;
```
Comment (kennykerr) [inline dia/src/main.rs:22]: This should accept the result of `as_os_str()` directly if you have the `alloc` and `std` features.

File: dia/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+        let session = dia_source.openSession()?;
+        let mut symbol_name = "sample::*\0\0".encode_utf16().collect::<Vec<u16>>();
+        let symbols = session.globalScope()?.findChildren(
+            SymTagFunction,
+            PWSTR(symbol_name.as_mut_ptr()),
```
Comment (kennykerr) [inline dia/src/main.rs:28]:
```suggestion
            "sample::*"
```

File: dia/src/main.rs (hunk 1 above)
Comment (riverar) [reply dia/src/main.rs]: Fixed.

**Comment by riverar** [comment]
Happy to leave this here until we land https://github.com/microsoft/windows-rs/pull/1336

File: dia/build.rs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+fn main() {
+    windows::core::build!({
+        Microsoft::Dia::*,
+        Windows::Win32::Foundation::*,
```
Comment (kennykerr) [inline dia/build.rs:4]:
```suggestion
```

Remove these - they're ignored - I should add a compiler error for this.

File: dia/build.rs [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+        Windows::Win32::System::Com::*,
+        Windows::Win32::System::SystemServices::*,
```
Comment (kennykerr) [inline dia/build.rs:6]:
```suggestion
```

Remove these - they're ignored - I should add a compiler error for this...

File: dia/src/main.rs [hunk 6]
```diff
@@ -0,0 +1,27 @@
+mod bindings {
+    windows::core::include_bindings!();
+}
+
+use bindings::Microsoft::Dia::*;
```
Comment (kennykerr) [inline dia/src/main.rs:5]: Maybe simpler if you remove the `bindings` module.

File: dia/build.rs (hunk 5 above)
Comment (riverar) [reply dia/build.rs:6]: Thanks, missed cleaning that up. Grr.

File: dia/.metadata/dia.manual.cs [hunk 7]
```diff
@@ -0,0 +1,52 @@
+using Windows.Win32.Foundation;
+
+namespace Microsoft.Dia
+{
+    public enum SymTag
```
Comment (kennykerr) [inline dia/.metadata/dia.manual.cs:5]: Still seems problematic that we need this.

**Review by kennykerr — APPROVED** [review]

File: dia/.metadata/dia.manual.cs (hunk 7 above)
Comment (riverar) [reply dia/.metadata/dia.manual.cs:5]: Yeah I hear ya. Here's a bit more context. https://github.com/microsoft/win32metadata/discussions/745.

**Comment by riverar** [comment]
Pivoting into a crate.

**Pull request closed without merging**